sixtyfps_runtime/interpreter/ffi.rs

```diff
@@ -582,11 +582,15 @@ impl Model for ModelAdaptorWrapper {
         self.0.row_count()
     }
 
-    fn row_data(&self, row: usize) -> Value {
-        unsafe {
-            let mut v = std::mem::MaybeUninit::<Value>::uninit();
-            self.0.row_data(row, v.as_mut_ptr() as *mut ValueOpaque);
-            v.assume_init()
+    fn row_data(&self, row: usize) -> Option<Value> {
+        if row >= self.row_count() {
+            None
+        } else {
+            unsafe {
+                let mut v = std::mem::MaybeUninit::<Value>::uninit();
+                self.0.row_data(row, v.as_mut_ptr() as *mut ValueOpaque);
+                Some(v.assume_init())
+            }
```

Comment:
This highlights that we probably should do the same change in the C++ Model side, and then the row_data function pointer would indicate to the caller if it returned something or not. So this would probably become the suggestion below, provided that `row_data` returns a `bool` and the C++ Model returns `std::optional`. That could be done in a separate change though?

```suggestion
    fn row_data(&self, row: usize) -> Option<Value> {
        unsafe {
            let mut v = std::mem::MaybeUninit::<Value>::uninit();
            if self.0.row_data(row, v.as_mut_ptr() as *mut ValueOpaque) {
                Some(v.assume_init())
            } else {
                None
            }
        }
```